gitlab/README.md

```diff
@@ -0,0 +1,142 @@
+The contents of the [event-display.yaml](samples/event-display.yaml)
+file are as follows:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+      - image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display
+```
+
+Enter the following command to create the service from [event-display.yaml](samples/event-display.yaml):
+
+```shell
+kubectl -n default apply -f samples/event-display.yaml
+```
+
+### Create GitLab Tokens
+
+1. Create a [personal access token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
+  which the GitLab source will use to register webhooks with the GitLab API.
+  The token must have an "api" access scope in order to create repository webhooks.
+  Also decide on a secret token that your code will use to authenticate the
+  incoming webhooks from GitLab.
+
+1. Update a secret defined in [secret.yaml](samples/secret.yaml):
+
+  ```yaml
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    name: gitlabsecret
+  type: Opaque
+  stringData:
+    accessToken: <personal_access_token_value>
+    secretToken: <random_string>
+  ```
+
+  Where `accessToken` is the personal access token created in step 1.
+  and `secretToken` (`random_string` above) is any token of your choosing.
+
+  Hint: you can generate a random _secretToken_ with:
+
+  ```shell
+  head -c 8 /dev/urandom | base64
+  ```
+
+1. Apply the gitlabsecret using `kubectl`.
+
+  ```shell
+  kubectl -n default apply -f samples/secret.yaml
+  ```
+
+### Create Event Source for GitLab Events
+
+1. In order to receive GitLab events, you have to create a concrete Event Source
+  for a specific namespace. Replace the `projectUrl` value in the file
+  `gitlabeventbinding.yaml` with your GitLab username and project name. For example,
+  if your repo URL is `https://gitlab.com/knative-examples/functions`
+  then use it as the value for `projectUrl`.
+
+  ```yaml
+  apiVersion: source.knative.dev/v1alpha1
+  kind: GitLabSource
+  metadata:
+    name: gitlabsource-sample
+  spec:
+    eventTypes:
+    - push_events
+    - issues_events
+    projectUrl: "<project url>"
+    accessToken:
+      secretKeyRef:
+        name: gitlabsecret
+        key: accessToken
+    secretToken:
+      secretKeyRef:
+        name: gitlabsecret
+        key: secretToken
+    sink:
+      apiVersion: serving.knative.dev/v1alpha1
+      kind: Service
+      name: gitlab-event-display
+  ```
+
+1. Apply the yaml file using `kubectl`:
+
+  ```shell
+  kubectl -n default apply -f samples/gitlabsource.yaml
+  ```
+
+### Verify
+
+Verify the GitLab webhook was created by looking at the list of
+webhooks under **Settings >> Integrations** in your GitLab project. A hook
+should be listed that points to your Knative cluster.
+
+Create a push event and check the logs of the Pod backing the
+`event-display`. You will see the GitLab event.
```

Comment:
(nit) How about adding a small example of the output to expect? (without any sensitive data 😉)